core-web/libs/ui/src/lib/directives/dot-state-restore/dot-state-restore.directive.ts

```diff
@@ -0,0 +1,48 @@
+import { AfterViewInit, Directive } from '@angular/core';
+
+import { TableState } from 'primeng/api';
+import { Table } from 'primeng/table';
+
+/**
+ * Directive to restore the state of a table component
+ * This need to be done because of this issue: https://github.com/primefaces/primeng/issues/10747
+ * if the sortField or sortOrder is defined in the table component, the table will not restore the state
+ */
+
+const enum StorageType {
+    Session = 'session',
+    Local = 'local'
+}
+
+@Directive({
+    selector: '[dotStateRestore]',
+    standalone: true
+})
+export class DotStateRestoreDirective implements AfterViewInit {
+    constructor(private table: Table) {}
```

Comment:
@hmoreras Use the `inject` function instead to handle injections in the constructor.

Reply:
Yes, use `inject()`